listeners.go

```diff
@@ -485,10 +491,73 @@ func ListenerUsage(network, addr string) int {
 	return count
 }
 
+// contextAndCancelFunc groups context and its cancelFunc
+type contextAndCancelFunc struct {
+	context.Context
+	context.CancelFunc
+}
+
+// sharedQuicTlsConfig manages GetConfigForClient
+type sharedQuicTlsConfig struct {
+	rmu           sync.RWMutex
+	tlsConfs      map[*tls.Config]contextAndCancelFunc
+	activeTlsConf *tls.Config
+}
+
+// newSqtc creates a new sharedQuicTlsConfig
+func newSqtc(tlsConfig *tls.Config) *sharedQuicTlsConfig {
```

Comment:
I think abbreviated names are okay for local variables, but for functions it's best to use the full name. `newSharedQuicTlsConfig`. Arguably it should probably be `sharedQUICTLSConfig` but that's a lot of uppercase letters in a row 😅 see https://google.github.io/styleguide/go/decisions.html#initialisms

Reply:
Agreed -- let's spell this out, and I prefer `sharedQUICTLSConfig` -- it's OK. :)